packages/nx/src/hasher/task-hasher.ts

```diff
@@ -482,7 +506,7 @@ class TaskHasherImpl {
           const externalDependencies = input['externalDependencies'];
           for (let dep of externalDependencies) {
             dep = this.findExternalDependencyNodeName(dep);
-            partialHashes.push(this.hashExternalDependency(dep));
+            partialHashes.push(this.hashExternalDependency(targetName, dep));
```

Comment:
Same here, source is the project name
```suggestion
            partialHashes.push(this.hashExternalDependency(projectName, dep));
```